extensions/amp-vr-image/0.1/amp-vr-image.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {addParamToUrl, assertHttpsUrl} from '../../../src/url';
+import {dev} from '../../../src/log';
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+import {loadPromise} from '../../../src/event-helper';
+
+/** @const */
+const TAG = 'amp-vr-image';
+
+class AmpVrImage extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    /** @const @private {boolean} */
+    this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
+    if (!this.isExperimentOn_) {
+      dev.warn(TAG, `TAG ${TAG} disabled`);
+      return;
+    }
+
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+
+    /** @private @const {number} */
+    this.width_ = getLengthNumeral(width);
+
+    /** @private @const {number} */
+    this.height_ = getLengthNumeral(height);
+
+    const imageSrc = assertHttpsUrl(this.element.getAttribute('src'),
+        this.element);
+    // TODO(dvoytenko): Consider recompiling and hosting viewer on the
+    // cdn.ampproject.org as an iframe viewer or even possibly compiling
+    // it as an AMP element.
+    let src = 'https://storage.googleapis.com/vrview/index.html';
+    src = addParamToUrl(src, 'image', imageSrc);
+    if (this.element.hasAttribute('stereo')) {
+      src = addParamToUrl(src, 'is_stereo', 'true');
+    }
+    const yaw = this.element.getAttribute('yaw');
+    if (yaw) {
+      src = addParamToUrl(src, 'start_yaw', yaw);
+    }
+    if (this.element.hasAttribute('yaw-only')) {
+      src = addParamToUrl(src, 'is_yaw_only', 'true');
+    }
+    /** @private @const {string} */
+    this.src_ = src;
+  }
+
+  /** @override */
+  preconnectCallback() {
```

Comment:
You mean a placeholder? Not easy, but there might be an option where we could center on the first half of the image in the center. I'll take a look.